packages/hardhat/contracts/strategy/CrabMigration.sol

```diff
@@ -19,18 +19,17 @@ import {StrategyMath} from "./base/StrategyMath.sol";
 
 /**
  * Migration Error Codes:
- * M1: Crab V2 Address already set
- * M2: Migration already happened
- * M3: Migration has not yet happened
- * M4: msg.sender is not Euler Mainnet Contract
- * M5: msg. sender cannot send ETH
- * M6: Can't withdraw more than you own
- * M7: Not enough ETH to repay the loan
+ * M1: Migration already happened
+ * M2: Migration has not yet happened
+ * M3: msg.sender is not Euler Mainnet Contract
+ * M4: msg.sender cannot send ETH
```

Comment:
This is for the receive right?
`    receive() external payable {
        require(msg.sender == weth || msg.sender == crabV1 || msg.sender == crabV2, "M4");
    }`

Reply:
Yeah it is. I think the meaning is that msg.sender (contract that is sending ETH) isnt allowed to, but maybe could be clearer.

Reply:
Agreed; now updated to "Can only receive ETH from weth, crabv1, or crabv2 contract".